src/DataValues/DecimalValue.php

```diff
@@ -93,33 +85,25 @@ private function assertNumberString( $number ) {
 	 * @throws InvalidArgumentException
 	 */
 	private function convertToDecimal( $number ) {
-		if ( !is_int( $number ) && !is_float( $number ) ) {
-			throw new InvalidArgumentException( '$number must be an int or float.' );
-		}
-
 		if ( $number === NAN || abs( $number ) === INF ) {
 			throw new InvalidArgumentException( '$number must not be NAN or INF.' );
 		}
 
 		if ( is_int( $number ) || ( $number === floor( $number ) ) ) {
 			$decimal = strval( abs( (int)$number ) );
 		} else {
-			$decimal = trim( number_format( abs( $number ), 100, '.', '' ), 0 );
+			$decimal = trim( number_format( abs( $number ), 100, '.', '' ), '0' );
 
 			if ( $decimal[0] === '.' ) {
 				$decimal = '0' . $decimal;
 			}
 
-			$last = strlen($decimal)-1;
-
-			if ( $decimal[$last] === '.' ) {
-				$decimal = $decimal . '0';
+			if ( substr( $decimal, -1 ) === '.' ) {
```

Comment:
In general, I agree with cleaning up code that is "close by". But I find myself trying to avoid this more and more, as I often find that it makes review harder. I have to look a lot closer to find the "meat" of the change. 

Also, some cleanup is debatable. In this case, I find the new version of the code (slightly) harder to read.